apps/emqx/src/emqx_cm.erl

```diff
@@ -502,6 +512,18 @@ lookup_channels(global, ClientId) ->
 lookup_channels(local, ClientId) ->
     [ChanPid || {_, ChanPid} <- ets:lookup(?CHAN_TAB, ClientId)].
 
+-spec lookup_client({clientid, emqx_types:clientid()} | {username, emqx_types:username()}) ->
+          [channel_info()].
+lookup_client({username, Username}) ->
+    MatchSpec = [{ {'_', #{clientinfo => #{username => '$1'}}, '_'}
```

Comment:
nit: is there a specific reason for not using `fun2ms`?

Reply:
This is copy-paste of the old code, I try to minimize business logic refactoring in these PRs to save time.